### ISSUE TITLE
Add a controller API to reflect whether force commit is finished

### ISSUE THREAD

**Issue by Jackie-Jiang** [apache/pinot#10069]
Force commit is executed asynchronously leveraging the Helix messages. Currently there is no way to tell whether force commit is finished or not. We want to add a controller API to indicate whether all the consuming segments are successfully committed.

There are 2 solutions I can think of:
1. Similar to tracking the reload status, we store the force commit request on the controller in a map, and use an uuid to look up the submitted force commit request. We can store the consuming segments when the force commit request is submitted, and then check if all of them turn into ONLINE state in the ideal state.
2. Do not track anything, and return the smallest of all the last committed segments' creation time. User can use this timestamp to tell whether all the segments are recently committed. For streams with some partition can be ended (e.g. Kinesis), it might need special handling so that the ended partition is excluded.

cc @jadami10 @siddharthteotia @mcvsubbu @sajjad-moradi for some discussion

**Comment by Jackie-Jiang** [comment]
@sajjad-moradi This is a good work around, and I think we should make this check automatic so that user doesn't need to implement this extra logic. This logic is very similar to approach 1 above, where we track the consuming segments when force commit is submitted, and then check if all the consuming segments are ONLINE in the force commit status check. This is very similar to how we track the reload status.

**Comment by jadami10** [comment]
hey Jackie, getting to this a little late.

I haven't looked at the code at all yet, but is there a way we could make these "async task" APIs more generically implemented in the backend. Things like force commit, rebalance, and other could really benefit from the same state tracking and polling features. I wonder if there's a way to make this easier in Pinot itself rather than trying to string together something to approximate the state you're looking for. I imagine minion must do something like this since it explicitly runs tasks whose status you can check

**Comment by Jackie-Jiang** [comment]
@jadami10 Yes, we are adding the general controller job concept to represent all these "aync tasks". Reload status being the first one, and we are adding force commit status as the second one in this [PR](https://github.com/apache/pinot/pull/10088)
